plugins/modules/wait_for_task.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# (c) 2023, Julien Godin <julien.godin@camptocamp.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: wait_for_task
+version_added: 3.13.0
+short_description: Wait for a task
+description:
+  - Wait for a task to finish
+author:
+  - "Julien Godin (@JGodin-C2C)"
+options:
+  task:
+    description:
+      - Task to wait for. Typical output of the resource_info module
+    required: true
+    type: dict
```

Comment:
I wonder if this should really be a dict like returned by `resource_info` vs a plain string with the ID, which we then feed like 
```python
module.wait_for_task({'id': module.foreman_params['task']})
```

@mdellweg @ehelms opinions?

Reply:
technically, I think we could even accept both via `type: raw` and then decide in code how to call `wait_for_task`

Reply:
I thought about this (leaving the decision to the code how to call `wait_for_task`), but not sure if that is a good idea. any additional input/opinion  would be great IMHO.

Reply:
Hey @ehelms sorry, didnt see your comment. 
Honestly , i do not think this module will be used a lot independently. It could cover some corner cases. But i think it will mostly be used in conjunction with other modules.

Reply:
What would it be? The task id? Its url? I tend towards a short identifier. If you come along with the output of a resource_info call, you can still use something like "result.task.href", no?

Reply:
No, it will simply be a task object.

Reply:
So let's say, all i have in my playbook is the task href, do i need to fake a dict to make this module work, while all the module will ever look at is the id/href inside the dict? That sounds like a strange interface.

Reply:
Even more so, we need to add a weird state attribute in that dict, and `wait_for_tasks` will try to interpret that. This sounds all wrong to me. All we should pass to the module is the id. And the module should first query the task state from the server.

Reply:
Ok. Thats two peoples in favor of the id as a parameter. I will implement.

Reply:
Okay ! 
It now uses the ID of the task and output the id + the task itself.